pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -82,12 +82,24 @@ func Diagnose(diagCfg diagnosis.Config) []diagnosis.Diagnosis {
 	for _, domainResolver := range domainResolvers {
 		// Go through all API Keys of a domain and send an HTTP request on each endpoint
 		for _, apiKey := range domainResolver.GetAPIKeys() {
-			for _, endpointInfo := range endpointsInfo {
-				domain, _ := domainResolver.Resolve(endpointInfo.Endpoint)
-				httpTraces := []string{}
-				ctx := httptrace.WithClientTrace(context.Background(), createDiagnoseTraces(&httpTraces))
-
-				statusCode, responseBody, logURL, err := sendHTTPRequestToEndpoint(ctx, client, domain, endpointInfo, apiKey)
+			for _, endpointInfo := range getEndpointsInfo(config.Datadog) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏You can move the call to `getEndpointsInfo` outside of the for loops

Reply:
Sorry but I don't really get why we want to call it before ?